pkg/controller/statefulset/stateful_set.go

```diff
@@ -130,6 +136,30 @@ func NewStatefulSetController(
 	ssc.podLister = podInformer.Lister()
 	ssc.podListerSynced = podInformer.Informer().HasSynced
 
+	if err := podInformer.Informer().AddIndexers(cache.Indexers{
+		StatefulSetIndex: func(obj interface{}) ([]string, error) {
```

Comment:
I'd prefer we avoid creating per-controller indexes. At the same time I see we've [added a generic index in replica set controller](https://github.com/kubernetes/kubernetes/blob/73f54b67b29d77601b0bd42ad8b4992925b9df47/pkg/controller/replicaset/replica_set.go#L167-L179). We guarantee [UID uniqueness within entire cluster](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/). How about re-using that generic index, in a similar way to what we've done in https://github.com/kubernetes/kubernetes/pull/130864. You'd basically introduce a new function `AddControllerUIDIndexer`, which you'll then use both in replica set controller and here.  The reason why we need to check for its existence is because admins can disable any controller. 

@wojtek-t wdyt?

Reply:
discussed on the other PR - we're on the same page